ansible/roles/test/files/ptftests/py3/copp_tests.py

```diff
@@ -492,6 +524,38 @@ def contruct_packet(self, port_number):
 
         return packet
 
+    def check_constraints(self, send_count, recv_count, time_delta_ms, rx_pps):
+        self.log("")
+        if self.has_trap:
+            self.log("Checking constraints (PolicyApplied):")
+            self.log(
+                "PPS_LIMIT_MIN (%d) <= rx_pps (%d) <= PPS_LIMIT_MAX (%d): %s" %
+                (int(self.PPS_LIMIT_MIN),
+                 int(rx_pps),
+                 int(self.PPS_LIMIT_MAX),
+                 str(self.PPS_LIMIT_MIN <= rx_pps <= self.PPS_LIMIT_MAX))
+            )
+            assert self.PPS_LIMIT_MIN <= rx_pps <= self.PPS_LIMIT_MAX, "rx_pps {}".format(rx_pps)
```

Comment:
@prabhataravind Can we add specific error messages when assertions fail? This will enhance the test summary by providing more detailed information if a test case fails. Currently, the summary only displays a generic 'AssertionError,' which doesn't offer a clear reason for the failure. This lack of detail makes it difficult to diagnose and triage the issue effectively.

Reply:
@ZhaohuiS, sure I can make the comment a bit more verbose.